arangodantic/tests/test_model.py

```diff
@@ -90,7 +90,7 @@ async def test_locking(identity_collection):
     identity = Identity(name="James Doe")
     await identity.save()
 
-    second_lock = CONF.lock(Identity.get_lock_name(identity.key_))
+    second_lock = Identity.get_lock(identity.key_)
```

Comment:
I guess the most convenient invokation in this case would be `identity.get_lock()`.. but we can't sacrifice the classmethod for that I suppose

Reply:
Yes, exactly. It would be really nice if Python would allow one to define both a class method and instance method with the same name.

Making it an instance method would make it impossible to use elegantly in the `lock_and_load` method which is a class method as well.

One of the use cases that I see for this class method is that you might want to create an instance with a predefined key, in such way that you obtain the lock for that key before you create the instance and still hold the lock while doing some other things afterwards. That could also be done by initiating the class first (without saving it) and getting the lock for it before attempting to save it. However that would require that you'd know all the required fields at the time you want to obtain the lock (or fill them with placeholder values). And this wouldn't work for the `lock_and_load` method, as it doesn't know what all fields would need to be filled.

So going with the now implemented solution. We could at some point later, if we see a need, implement that with some other name. In the meantime, I think the `lock_and_load` and `lock_and_reload` should cover most of the practical use cases and manually giving the key a somewhat OK way to deal with other cases.